src/include/session.h

```diff
@@ -53,6 +53,12 @@ typedef TAILQ_HEAD(__wt_cursor_list, __wt_cursor) WT_CURSOR_LIST;
 /* Maximum number of buckets to visit during a regular cursor sweep. */
 #define WT_SESSION_CURSOR_SWEEP_MAX 64
 
+/* Invalid session ID. */
+#define WT_SESSION_ID_INVALID 0xffffffff
+
+/* A fake session ID for when we need to refer to a session that is actually NULL. */
+#define WT_SESSION_ID_NULL 0xfffffffe
```

Comment:
Given that you explicitly set the id to 64 bits, should these really be something like `include/log.h` where we have:
#define WT_LOG_SLOT_RESERVED 0x8000000000000000LL /* Reserved states */
#define WT_LOG_SLOT_MASK_OFF 0x3fffffffffffffffLL

I.e. explicitly use 64 bits in the id. And if you're creating a special value set them from the high bits in case we ever have a need to add a mask or bits, etc.

Reply:
Our session IDs are actually 32-bit, even as the corresponding field in the `__wt_session_impl` struct is `uint32_t`, so this is meant to be a 32-bit constant in case it might become somewhere else.

I declared the session ID field in the mutex as `uint64_t` so that everything is properly 8-byte memory aligned, but now that I'm looking at the `__wt_spinlock` struct again, this is probably pointless as everything after that is `int16_t` or `int8_t`. Maybe I should just change the session ID field in `__wt_spinlock` to `uint32_t`.

Reply:
You could use uint32_t and if alignment is needed add a `uint8_t unused[4];` field.  I think if everything is `uint32_t` it would be best to have this also be.

Reply:
Sounds good, I'll keep it in mind for next time. I don't think we need the alignment this time—now that I think about it, I'm not quite sure why I put it in there the first time.